src/main/kotlin/com/openlattice/shuttle/api/ShuttleApi.kt

```diff
@@ -0,0 +1,104 @@
+package com.openlattice.shuttle.api
+
+import com.openlattice.shuttle.control.Integration
+import com.openlattice.shuttle.control.IntegrationJob
+import com.openlattice.shuttle.control.IntegrationStatus
+import com.openlattice.shuttle.control.IntegrationUpdate
+import retrofit2.http.*
+import java.util.*
+
+
+const val SERVICE = "/shuttle"
+const val CONTROLLER = "/integration"
+const val BASE = SERVICE + CONTROLLER
+
+const val DEFINITION_PATH = "/definition"
+const val STATUS_PATH = "/status"
+
+const val INTEGRATION_NAME = "integrationName"
+const val INTEGRATION_NAME_PATH = "/{$INTEGRATION_NAME}"
+const val INTEGRATION_KEY = "integrationKey"
+const val INTEGRATION_KEY_PATH = "/{$INTEGRATION_KEY}"
+const val JOB_ID = "jobId"
+const val JOB_ID_PATH = "/{$JOB_ID}"
+const val CALLBACK = "callbackUrl"
+const val CALLBACK_PATH = "/{$CALLBACK}"
+
+interface ShuttleApi {
+
+    /**
+     * Enqueues an integration on Shuttle Server for a given integration
+     * @param integrationName the name of the integration to be run
+     * @param integrationKey the unique id used to authenticate an integration run
```

Comment:
Needs `@param callbackUrl`
Also what is callbackUrl for/what is the contract of such a thing? (I've never implemented such a thing)

Reply:
It's to signal that the integration has finished. I've never implemented such a thing either so we're learning.